src/slices/MainSlice.js

```diff
@@ -0,0 +1,176 @@
+
+  token = contractForReserve({ bond: BONDS.frax, networkID, provider });
+  let fraxAmount = await token.balanceOf(addresses[networkID].TREASURY_ADDRESS);
+
+  token = contractForReserve({ bond: BONDS.eth, networkID, provider });
+  let ethAmount = await token.balanceOf(addresses[networkID].TREASURY_ADDRESS);
+
+  token = contractForReserve({ bond: BONDS.ohm_dai, networkID, provider });
+  let ohmDaiAmount = await token.balanceOf(addresses[networkID].TREASURY_ADDRESS);
+  let valuation = await bondCalculator.valuation(addressForAsset({ bond: BONDS.ohm_dai, networkID }), ohmDaiAmount);
+  let markdown = await bondCalculator.markdown(addressForAsset({ bond: BONDS.ohm_dai, networkID }));
+  let ohmDaiUSD = (valuation / Math.pow(10, 9)) * (markdown / Math.pow(10, 18));
+
+  token = contractForReserve({ bond: BONDS.ohm_frax, networkID, provider });
+  let ohmFraxAmount = await token.balanceOf(addresses[networkID].TREASURY_ADDRESS);
+  valuation = await bondCalculator.valuation(addressForAsset({ bond: BONDS.ohm_frax, networkID }), ohmFraxAmount);
+  markdown = await bondCalculator.markdown(addressForAsset({ bond: BONDS.ohm_frax, networkID }));
+  let ohmFraxUSD = (valuation / Math.pow(10, 9)) * (markdown / Math.pow(10, 18));
+
+  const treasuryBalance =
+    daiAmount / Math.pow(10, 18) +
+    fraxAmount / Math.pow(10, 18) +
+    (ethAmount / Math.pow(10, 18)) * ethPrice +
+    ohmDaiUSD +
+    ohmFraxUSD;
+
+  // Calculating staking
+  const epoch = await stakingContract.epoch();
+  const stakingReward = epoch.distribute;
+  const circ = await sohmMainContract.circulatingSupply();
+  const stakingRebase = stakingReward / circ;
+  const fiveDayRate = Math.pow(1 + stakingRebase, 5 * 3) - 1;
+  const stakingAPY = Math.pow(1 + stakingRebase, 365 * 3) - 1;
+
+  // TODO: remove this legacy shit
+  const oldStakingReward = await oldStakingContract.ohmToDistributeNextEpoch();
+  const oldCircSupply = await sohmOldContract.circulatingSupply();
+
+  const oldStakingRebase = oldStakingReward / oldCircSupply;
+  const oldStakingAPY = Math.pow(1 + oldStakingRebase, 365 * 3) - 1;
+
+  // Current index
+  const currentIndex = await stakingContract.index();
+
+  return {
+    currentIndex: ethers.utils.formatUnits(currentIndex, "gwei"),
+    currentBlock,
+    fiveDayRate,
+    treasuryBalance,
+    stakingAPY,
+    stakingTVL,
+    oldStakingAPY,
+    stakingRebase,
+    marketCap,
+    marketPrice,
+    circSupply,
+    totalSupply,
+  };
+});
+
+export const getFraxData = createAsyncThunk("app/getFraxData", async () => {
+  const resp = await axios.get("https://api.frax.finance/combineddata/");
+  return {
+    type: Actions.FETCH_FRAX_SUCCESS,
+    payload: resp.data && resp.data.liq_staking && resp.data.liq_staking["Uniswap FRAX/OHM"],
+  };
+});
+
+const appSlice = createSlice({
+  name: "app",
+  initialState,
+  reducers: {
+    fetchAppSuccess(state, action) {
+      setAll(state, action.payload);
+    },
+  },
+  extraReducers: builder => {
```

Comment:
For this Slice, why do we need to put the things in "extraReducer". It looks like they're all dependent on loadAppDetails anyway. Unless im missing something with how the pending/fullfilled/idle states get updated. 

my assumption is that the state.loading would have something.

Reply:
oh nvm. i think i get it. the loadAppDetails, is an async action, and the appSlice is dependent on its completion.

Reply:
yeah the extra reducer watches the thunk pulling the data and we get to change state.pending and then fill in the data when it resolves